migrate.sh

```diff
@@ -42,18 +42,48 @@ echo "Source Directory: $source_dir"
 echo "Destination Directory: $destination_dir"
 echo "" # Blank line to separate from any failure output
 
-# Convert source and destination directories to absolute paths
-source_dir=$(readlink -f "$source_dir")
-destination_dir=$(readlink -f "$destination_dir")
+is_absolute_path() {
+    case "$1" in
+        /*) return 0 ;;  # Absolute path
+        ~/*) return 0 ;; # Absolute path with home directory
+        *) return 1 ;;   # Relative path
+    esac
+}
+
+# Convert source and destination directory to absolute path if relative
+if ! is_absolute_path "$source_dir"; then
+    source_dir=$(readlink -f "$source_dir") || { echo "Error: Failed to resolve source directory path, directory may not exist"; exit 1; }
+fi
+
+cel2_migration_tool_image="us-west1-docker.pkg.dev/devopsre/celo-blockchain-public/cel2-migration-tool"
+cel2_migration_tool_tag="5682b80ec60c47f582c6af8aa085ae6f9048d801"
+
+# Run check-db continuity script to ensure source db has no data gaps
+if docker run --platform=linux/amd64 -it --rm \
+    -v "${source_dir}/celo/chaindata:/old-db" \
+    "${cel2_migration_tool_image}:${cel2_migration_tool_tag}" \
+    check-db \
+      --db-path /old-db \
+      --fail-fast; then
+    printf "\033[0;32mDB check completed successfully. No gaps / missing data detected.\033[0m\n"
+else
+    printf "\033[0;31mDB check failed with exit code $?. If the logs indicate that the db is missing data, please retry with another source db. You can visit https://docs.celo.org/cel2/operators/migrate-node for instructions on how to check whether a db has missing data.\033[0m\n"
+    exit $?
+fi
 
 # Ensure destination directory exists for chaindata
-mkdir -p  "${destination_dir}/geth"
+mkdir -p  "${destination_dir}/geth/chaindata"
+
+# Convert destination directory to absolute path if relative
+if ! is_absolute_path "$destination_dir"; then
+    destination_dir=$(readlink -f "$destination_dir") || { echo "Error: Failed to resolve destination directory path"; exit 1; }
```

Comment:
Given that `readlink -f` only fails when faced with non existent directories, this check seems redundant since we just made the directory above.